packages/rust/server/src/services/email.rs

```diff
@@ -1,10 +1,95 @@
+use crate::config::CONFIG;
+use sendgrid::SGClient;
+use sendgrid::SendgridError;
+use sendgrid::{Destination, Mail};
+
 #[derive(Default, Clone, Copy)]
 pub struct EmailService {}
+pub struct MockEmailService {}
+
+pub trait Email_ {
```

Comment:
trait should be called EmailService, you should have two structs SendgridEmailService and PrinterEmailService or something like that. Trait also needs to be implemented for all routes.